Framework/PythonInterface/plugins/algorithms/FindGoniometerFromUB.py

```diff
@@ -0,0 +1,293 @@
+                "(perpendicular to ki)")
+        self.declareProperty(
+            name="OmegaHand",
+            defaultValue=1,
+            validator=IntListValidator([-1, 1]),
+            direction=Direction.Input,
+            doc="Handedness for the omega angles (rotation around +ve Z) in the log file (1 for ccw/RH, -1 for cw/LH)")
+        # output
+        self.declareProperty(
+            ITableWorkspaceProperty(
+                name="OutputTable",
+                defaultValue="",
+                validator=StringMandatoryValidator(),
+                direction=Direction.Output),
+            doc="Output table to display goniometer axis and angles for runs with UBs that can share common indexing")
+
+    def validateInputs(self):
+        issues = dict()
+        # UBpath must contain more than one run
+        ubFiles = self.getProperty('UBfiles').value
+        if len(ubFiles) < 2:
+            issues['UBfiles'] = "At least two UB files are required"
+        return issues
+
+    def PyExec(self):
+
+        # setup progress bar
+        prog_reporter = Progress(self, start=0.0, end=1.0, nreports=3)
+        # Get input
+        ubFiles = self.getProperty('UBfiles').value
+        chiRef = self.getProperty('Chi').value
+        chiTol = self.getProperty('ChiTol').value
+        phiTol = self.getProperty('PhiTol').value
+        phiHand = self.getProperty('PhiHand').value
+        dOmega = self.getProperty('DOmega').value
+        omegaHand = self.getProperty('OmegaHand').value
+        # rotation matrices work in right-handed coordinate system so the handedness
+
+        # self.log().information(msg)
+        # MultipleFileProperty -> UB matrix (first one for reference)
+
+        # update progress
+        prog_reporter.report(1, "Load Logs and UBs")
+
+        matUB = []  # container to hold UB matrix arrays
+        omega = np.zeros(len(ubFiles))  # rot around vertical axis (assumed to be correct)
+        phiRef = np.zeros(len(ubFiles))  # rot around gonio axis (needs to be refined)
+        for irun in range(0, len(ubFiles)):
+            # get rotation angles from logs (handedness given in input)
+            # these rotation matrices are defined in right-handed coordinate system (i.e. omegaHand = 1 etc.)
+            tmpWS = CreateSampleWorkspace(StoreInADS=False)
+            _, fname = path.split(ubFiles[irun])
+            dataPath = FileFinder.findRuns(fname.split('.mat')[0])[0]
+            if dataPath[-4:] == ".raw":
+                # assume log is kept separately in a .log file with same path
+                LoadLog(Workspace=tmpWS, Filename="".join(dataPath[:-4] + '.log'))
+            elif dataPath[-4:] == '.nxs':
+                # logs are kept with data in nexus file
+                LoadNexusLogs(Workspace=tmpWS, Filename=dataPath)
+            # read omega and phi (in RH coords)
+            omega[irun] = omegaHand * tmpWS.getRun().getLogData('ccr_pos').value[0]
```

Comment:
Can we avoid hard coding the log names as other instruments will undoubtedly have different names for their logs. Perhaps we could have 2 extra input parameters for the names of the logs containing the Omega and Phi angles?